### ISSUE TITLE
Flash of weird colors on game startup

### ISSUE THREAD

**Issue by lekoder** [godotengine/godot#22463]
**Godot version:**
3.1.alpha 5a03d50

**OS/device including version:**
Windows 10
GPU: NVIDIA GeForce GTX 1070 MQ, driver: 411.63
GPU: Interl UHD Graphics 630

**Issue description:**
When game starts, just as first scene renders, it appears for a brief second frozen, with distorted colors:
![Flash of color](https://thumbs.gfycat.com/TastyShortFlycatcher-size_restricted.gif)

**Steps to reproduce:**
Unable to reproduce in minimal project. Issue can be observed with (demo version of ΔV: Rings of Saturn 0.20.0, available on itch.io)[https://koder.itch.io/dv-rings-of-saturn-demo] - observed with windows builds.

It might be related to loading resources and auto-exposure of `WorldEnvironment`. Going to work around it by providing blank/black initial scene.

**Comment by akien-mga** [comment]
I doubt this is an engine issue, this must be related to the way you cache resources in the title scene and maybe add them to the scene tree (thus making them potentially visible) before the GUI is initialized on top of them.

**Comment by Xrayez** [comment]
> It might be related to loading resources and auto-exposure of WorldEnvironment. Going to work around it by providing blank/black initial scene.

I can confirm that, specifically the auto-exposure of WorldEnvironment. I often experience this effect when the debugger catches some error immediately after scene startup, or even worse when the frame rate drops significantly.

**Comment by lekoder** [comment]
Isolated to [this minial prokject](https://github.com/lekoder/godot-issue-flash-of-color). Took a while since I was stripping down my production code.

It is triggered in my case by:
```
func _ready():
	OS.window_per_pixel_transparency_enabled = false
```

**Comment by akien-mga** [comment]
CC @bruvzg

**Comment by lekoder** [comment]
Please note that I seen this effect without disabling per pixel transparency. It is a trigger, not a cause.

**Comment by akien-mga** [comment]
Do you see it if you don't use per pixel transparency at all?

**Comment by lekoder** [comment]
Yes. One way to trigger it without per-pixel transparency is to use Multi-Threaded or Single-Unsafe rendering model.

I have bug report of these artifacts happening when recording dV with OBS, but I was unable to replicate.

**Comment by lekoder** [comment]
I gave the issue some thought and I have a hypothesis. There are two separate things happening:

1. The `WorldEnvironment` Auto Expose starts up with maximum exposition in the first frame and adjusts to actual exposition in subsequent frames. This usually goes unnoticed, however...
2. Some things, like changing per pixel transparency, cause rendering engine to freeze for ~0.5-1 second. If it happens on the first frame, it exposes the problem.

**Comment by lekoder** [comment]
@Xrayez this issue was fixed for me with install of [KB4462933 OS Build 17134.376](https://support.microsoft.com/pl-pl/help/4462933/windows-10-update-kb4462933) via Windows Upgrade. Could you cross-check?

**Comment by Xrayez** [comment]
@lekoder just installed mentioned update manually, this hasn't fixed the issue for me, maybe it was due to graphics driver update?

Btw, this auto-exposure issue can be easily reproduced with:
```
Project Settings > Application > Run > Frame Delay Msec > set to 1000 (one second)
```
> Some things, like changing per pixel transparency, cause rendering engine to freeze for ~0.5-1 second. If it happens on the first frame, it exposes the problem.

This simulates high CPU load and introduces enough delay.

**Comment by lekoder** [comment]
> maybe it was due to graphics driver update

Might be, those things are tough to replicate if we don't know what exactly is triggering it - and I don't pay close attention to it, as I worked around it in my game. I assume the underlying problem is freezing for significant fractions of a second.

**Comment by akien-mga** [comment]
So it seems to be driver-specific in the end?

**Comment by Xrayez** [comment]
Not sure, doesn't seem so, this is still reproducible in master in my own project.

Upon further testing, I managed to create a minimal project:
[auto-exposure.zip](https://github.com/godotengine/godot/files/2791669/auto-exposure.zip)

The sprite will appear white if enough delay is introduced:

![auto-exposure-flicker](https://user-images.githubusercontent.com/17108460/51678422-0dbd6a80-1fe5-11e9-8ca3-3a0ff60e9f13.png)

By enough I mean that the delay has to be as twice longer than the auto exposure speed! The default auto exposure speed is 0.5 sec, so 1 sec delay is the minimum time to cause the issue.

**Comment by Xrayez** [comment]
Setting `OS.window_per_pixel_transparency_enabled` is indeed not the cause, but it triggers the flickering by introducing the delay, and there are numerous other ways to trigger the delay, so maybe rename the issue.

**Comment by akien-mga** [comment]
Still reproducible in the current master branch with @Xrayez's minimal project.

**Comment by akien-mga** [comment]
It seems to only happen with the GLES3 driver, not GLES2.
CC @clayjohn.

**Comment by KoBeWi** [comment]
Tested the project in 3.2.4 beta3 and the Environment seems to affect 2D canvas
![NefwxPpEQ7](https://user-images.githubusercontent.com/2223172/101705565-7c93f780-3a87-11eb-9bfb-ef404369b729.gif)
I'm pretty sure it's not supposed to be like that. Same thing in master.
